### PR TITLE
Add BSP for Arduino RP2040 Nano Connect

### PR DESCRIPTION
**Pull request by splicedbread**
BSP for Arduino's RP2040 Nano Connect. Follows rp-pico board example. Peripherals are labeled in doc, but not included in this.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -16,6 +16,7 @@ members = [
     "boards/rp-pico",
     "boards/solderparty-rp2040-stamp",
     "boards/sparkfun-pro-micro-rp2040",
+    "boards/arduino_nano_connect",
```
Comment (9names) [inline Cargo.toml:19]: This list was in alphabetical order - can you put this between adafruit-trinkey-qt2040 and pimoroni-pico-explorer?

File: boards/arduino_nano_connect/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+If you get an error about not being able to find `elf2uf2-rs`, try:
+
+```console
+$ cargo install elf2uf2-rs
+```
+then try repeating the `cargo run` command above.
+
+### [nano_blinky](./examples/nano_blinky.rs)
+
+Flashes the Pico's on-board LED on and off.
```
Comment (9names) [inline boards/arduino_nano_connect/README.md:61]: Not a Pico

**Review by 9names — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (splicedbread) [reply Cargo.toml:19]: Can do

File: boards/arduino_nano_connect/README.md (hunk 2 above)
Comment (splicedbread) [reply boards/arduino_nano_connect/README.md]: Fixed typo in most recent commit below

**Review by 9names — APPROVED** [review]
Looks good to me, thanks!

File: README.md [hunk 3]
```diff
@@ -250,6 +250,17 @@ RP2040 chip according to how it is connected up on the Thing Plus RP2040.
 [Sparkfun Thing Plus RP2040]: https://www.sparkfun.com/products/17745
 [sparkfun-thing-plus-rp2040]: https://github.com/rp-rs/rp-hal/tree/main/boards/sparkfun-thing-plus-rp2040
 
+### [arduino_nano_connect] - Board Support for the [Ardunio Nano RP2040 Connect]
+
+You should include this crate if you are writing code that you want to run on
+an [Arduino Nano Connect] - a development pcb with shortwave communication, IMU, and BLE package.
```
[the remark this answers is not in the document]
Comment (9names) [reply README.md:256]: So you know what I'm talking about: Currently you have
[Arduino Nano Connect]
[Ardunio Nano RP2040 Connect]

These are definitely not the same.

[the remark this answers is not in the document]
Comment (splicedbread) [reply README.md:256]: Ahh, I had modified those earlier without updating the reference. Fixing now.

**Review by 9names — CHANGES_REQUESTED** [review]
Still a bunch of misspellings of arduino in here

File: boards/arduino_nano_connect/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
+# [arduino_nano_connect] - Board Support for the [Ardunio Nano RP2040 Connect]
```
[the remark this answers is not in the document]
Comment (splicedbread) [reply boards/arduino_nano_connect/README.md]: Went through with rgrep that was recommended and then found and replaced all instances of the incorrect spelling.

**Comment by 9names** [comment]
There's still one link that is broken - I'll fix it up and then we'll merge this.
Thanks for the PR and being patient, @splicedbread :+1:

**Review by 9names — APPROVED** [review]

**Pull request merged**